app/controllers/test_messages_controller.rb

```diff
@@ -1,27 +1,39 @@
 class TestMessagesController < ApplicationController
   def new
-    @message = ScheduledMessage.find(params[:scheduled_message_id])
+    if params[:scheduled_message_id]
+      @message = ScheduledMessage.find(params[:scheduled_message_id])
+      @url = scheduled_message_test_messages_path(@message)
+    else
+      @message = Message.find(params[:message_id])
+      @url = message_test_messages_path(@message)
+    end
   end
 
   def create
-    message = ScheduledMessage.find(params[:scheduled_message_id])
     employee = Employee.find_by(slack_username: slack_username)
 
-    if !employee.slack_channel_id.nil? || !employee.slack_user_id.nil?
+    if employee && (employee.slack_channel_id || employee.slack_user_id)
```

Comment:
ah yes we have some definite improvement to do in this controller! (pre-your contributions)